command/test_utils.go

```diff
@@ -32,6 +33,25 @@ func TestMetaFile(t *testing.T) Meta {
 	}
 }
 
+// GetStdoutAndErrFromTestMeta extracts stdout/stderr from a Meta created by TestMetaFile
+func GetStdoutAndErrFromTestMeta(t *testing.T, m Meta) ([]byte, []byte) {
```

Comment:
Seeing as the ui.Writer and ui.ErrorWriter are already bytes.Buffer specifically for capturing the test output we don't really need to copy into other buffers. Nothing wrong with this it is just extra work to validate against. 

That said there is a function already for returning string values for both ui.Writer and ui.ErrorWriter I recommend using that function and making the suggestions I left above. https://github.com/hashicorp/packer/blob/6ee6091aa9fee2f03f57d9fa917dbb6f18b39da5/command/command_test.go#L25

Reply:
Just want to circle back here, as my initial comment can use some context. Since the TestMeta Writers are for testing and already bytes.Buffer we can convert and access them directly where needed. The function you have makes sense and the copy does protect against unexpected changes. So it is a good convenience function to have if we need to pas the the outputs around as bytes. I suspect that we are mostly interested in the string version of the Writers, which is the reason I'm recommending the outputCommand function. Yes, it is weirdly named so renaming as we discussed is a good way to go.